dGame/dComponents/PropertyComponent.cpp

```diff
@@ -2,10 +2,8 @@
 #include "GameMessages.h"
 #include "dZoneManager.h"
 
-PropertyComponent::PropertyComponent(Entity* parent) : Component(parent) {
-	m_PropertyName = parent->GetVar<std::string>(u"propertyName");
-	m_PropertyState = new PropertyState();
+PropertyComponent::PropertyComponent(Entity* parentEntity)
+	: Component{ parentEntity }
+	, m_PropertyName{ parentEntity->GetVar<std::string>(u"propertyName") }
+	, m_PropertyState{ std::make_unique<PropertyState>() } {
```

Comment:
personally I dont like using chained constructors like this for called code.  It would be better to have this in the body of the constructor instead.